common/web/types/src/kmx/kmx-plus-builder/build-disp.ts

```diff
@@ -0,0 +1,54 @@
+import { constants } from '@keymanapp/ldml-keyboard-constants';
+import { KMXPlusData } from "../kmx-plus.js";
+import { build_strs_index, BUILDER_STRS } from "./build-strs.js";
+import { BUILDER_SECTION } from './builder-section.js';
+
+/* ------------------------------------------------------------------
+ * vkey section
+   ------------------------------------------------------------------ */
+
+/**
+ * Builder for the 'vkey' section
+ */
+interface BUILDER_DISP_ITEM {
+  to: number;
+  display: number;
+};
+
+export interface BUILDER_DISP extends BUILDER_SECTION {
+  count: number;
+  baseCharacter: number;
+  reserved0: number;
+  reserved1: number;
+  reserved2: number;
+  reserved3: number;
+  items: BUILDER_DISP_ITEM[];
+};
+
+export function build_disp(kmxplus: KMXPlusData, sect_strs: BUILDER_STRS): BUILDER_DISP {
+  if(!kmxplus.disp?.disps?.length && !kmxplus.disp?.baseCharacter?.value) {
```

Comment:
```suggestion
  if(!kmxplus.disp?.disps?.length || !kmxplus.disp?.baseCharacter?.value) {
```

I am not sure that this is correct per spec. I see that on, e.g., line 35 and line 38 we depend on both of these being valid, so either we require them both or we need to handle them being nullish.

Reply:
Trying to understand this from the spec.  `count` could be zero in the kmx+. `disp` is always defined. `disps` is undefined if the `<display>` element isn't present at all. I've changed `Disp.compile()` to always have `disps` be `[]` even if ther are no disps, for robustness.   kmxplus.disp.baseCharacter is always present but may have value=''

So in conclusion i'll take out the questionmarks, because the compiler should always create at least:

```
//Disp
{
  baseCharacter: { value: ""},
  disps: []
}
```